azext_edge/edge/providers/orchestration/connected_cluster.py

```diff
@@ -38,6 +90,11 @@ def resource(self) -> dict:
     def location(self) -> str:
         return self.resource["location"]
 
+    @property
+    def connected(self) -> bool:
+        properties = self.resource.get("properties", {})
+        return "connectivityStatus" in properties and properties["connectivityStatus"].lower() == "connected"
```

Comment:
properties.get("connectivityStatus", "").lower() == "connected"

shorter and imo not more complicated?

Reply:
Yet slower

Reply:
what... how?

Reply:
"connectivityStatus in properties" is O(1) + properties["connectivityStatus"] is O(1) .... with properties.get("connectivityStatus", "") first the python runtime must lookup the attribute .get and execute, which will push and pop a frame to the stack, adding more compute expense compared to the former pattern.